ops/model.py

```diff
@@ -1324,7 +1368,227 @@ def list_files(self, path: str, *, pattern: str = None,
+            container.pull_path(['/foo/bar', '/quux.txt'], '/dst')
+            # Destination results: /dst/bar/baz.txt, /dst/quux.txt
+
+        Args:
+            source_path: A single path or list of paths to pull from the remote system.  The
+                paths can be either a file or a directory but must be absolute paths.  If
+                source_path is a directory, the directory base name is attached to the destination
+                directory - i.e.  the source path directory is placed inside the destination
+                directory.  If a source path ends with a trailing "/*" it will have its *contents*
+                placed inside the destination directory.
+            dest_dir: Local destination directory inside which the source dir/files will be
+                placed.
+        """
+        if os.name == 'nt':
+            raise RuntimeError('Container.pull_path is not supported on Windows-based systems')
+
+        source_paths = source_path if isinstance(source_path, list) else [source_path]
+        source_paths = [str(p) for p in source_paths]
+        dest_dir = str(dest_dir)
+
+        errors = []
+        for source_path in source_paths:
+            try:
+                for f in Container._list_recursive(self.list_files, source_path):
+                    srcpath, dstpath = f.path, self._build_destpath(f.path, source_path, dest_dir)
+                    os.makedirs(os.path.dirname(dstpath), exist_ok=True)
+                    with self.pull(srcpath, encoding=None) as src:
+                        with open(dstpath, 'wb') as dst:
+                            shutil.copyfileobj(src, dst)
+            except Exception as err:
+                errors.append((source_path, err))
+        if len(errors) > 0:
+            raise MultiPushPullError('failed to pull one or more files', errors)
+
+    @staticmethod
+    def _build_fileinfo(path: str) -> ops.pebble.FileInfo:
+        """Constructs a FileInfo object by stat'ing a local path."""
+        if os.path.islink(path):
+            ftype = pebble.FileType.SYMLINK
+        elif os.path.isdir(path):
+            ftype = pebble.FileType.DIRECTORY
+        elif os.path.isfile(path):
+            ftype = pebble.FileType.FILE
+        else:
+            ftype = pebble.FileType.UNKNOWN
+
+        import grp
+        import pwd
+        info = os.lstat(path)
+        return pebble.FileInfo(
+            path=path,
+            name=os.path.basename(path),
+            type=ftype,
+            size=info.st_size,
+            permissions=stat.S_IMODE(info.st_mode),
+            last_modified=info.st_mtime,
+            user_id=info.st_uid,
+            user=pwd.getpwuid(info.st_uid).pw_name,
+            group_id=info.st_gid,
+            group=grp.getgrgid(info.st_gid).gr_name)
+
+    @staticmethod
+    def _list_recursive(list_func, path: str):
+        """Recursively lists all files under path using the given list_func.
+
+        list_func
```

Comment:
This docstring needs an "Args:" heading to be consistent with the goog style guide :)

Reply:
fixed.